### ISSUE TITLE
Eyeball. More distinction between hide and visible

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-ui-icons#133]
At the 16px size in an interface, the hide and visible icons can look too similar. Would like to see more distinction between these two.

![eyeball](https://user-images.githubusercontent.com/12503298/84799431-1d11cf00-afb1-11ea-812e-da70d25c7ad3.gif)

One idea would be something like Figma's approach:
![figma eyeball](https://user-images.githubusercontent.com/12503298/84799660-7417a400-afb1-11ea-9da1-b56af496e563.gif)

@KatelynSeitz @johnmgriffith

**Comment by asangma** [comment]
Some were maybe reluctant about the lashes, but my armchair psychologist says their reluctance is gender bias. 🤔

**Comment by eriklharper** [comment]
I would expect this state of the eye to appear darker and/or bolder in color but it looks like the colors stay the same between the two states.  Perhaps just darkening the eye might address that?  

![image](https://user-images.githubusercontent.com/821864/84799985-f43e0980-afb1-11ea-8254-d9485971896e.png)

**Comment by asangma** [comment]
@eriklharper We actually started there, but it can appear to be disabled in the lighter state. And...this is the fun one: we sometimes have `user initiated hiding` along with `scale dependent hiding`, and we're using a lighter color on the entire element to indicate `scale dependent hiding`.

Fun times. 😆

**Comment by eriklharper** [comment]
Ah, yes, fun fun fun!  #uxishard

**Comment by asangma** [comment]
@KatelynSeitz @johnmgriffith Let me know if you need any additional details on this or if you have any feedback or concerns before I create a request. :)

**Comment by KatelynSeitz** [comment]
I am not against the suggested closed eyelid solution above. Would a new solution replace `view-hide` or is there a use case where we still need a crossed out eye?

That being said, I am curious about the solution below. Im my experience the most distinct and straightforward view/hide interaction is when the eye icon is either visible or empty. 
![Sketch](https://user-images.githubusercontent.com/29716057/84960063-8ed34100-b0b5-11ea-8a91-20659c6ddac5.gif)
![Illustrator](https://user-images.githubusercontent.com/29716057/84960469-8cbdb200-b0b6-11ea-8ac3-fbe5c22ad143.gif)

I find this to be true for a lock/unlock interaction as well, but I am guessing, based on your screenshot, the lock state has a closed lock icon and the unlock state changes the icon to an open lock?

**Comment by asangma** [comment]
@KatelynSeitz Indeed. The top example is what we currently employ. And it is somewhat straight forward if the user sees the icon as a verb, i.e. "hide this".

We've gotten feedback that some users see the icon as the adjective/descriptor "hidden" which is accurate and good on layers that are hidden. However, when those same users see the "hidden" icon while hovering over a **visible** layer, they find it confusing.

But whether crossed or closed, and this use case aside, the two icons at 16px can be hard to discern especially for those who've been around the sun a bunch. :)

**Comment by KatelynSeitz** [comment]
In all of these examples, I think you are right, it is not the change of icon (open to closed/ crossed, outline to filled etc.) that does the heavy lifting to indicate a visible or hidden state to the user (especially at 16px); it is actually the UI treatment. In your Figma gif above, the change in icon alone would not be obvious, but the text and icon color change is what makes the interaction more successful. I know you mentioned light gray might not be the exact solution for your use case, but either way, I think the icon solution needs to be paired with a UI treatment unless it is a single icon that toggles on and goes away for off. 

Options like the closed eyelid might work specifically during a toggle experience, but if it doesn't really solve the visibility issue at 16px, I might be more inclined to keep the crossed out eye. It is a more established icon and can stand on its own outside of the toggle interaction, where the closed eyelid would not as work well on its own for concepts like these. 

![image](https://user-images.githubusercontent.com/29716057/85077180-8abb2800-b176-11ea-9aba-e491d2b41c10.png)

**Comment by KatelynSeitz** [comment]
Griff and I were looking at this example in Scene Viewer to mock up the closed eyelid and it makes sense when you animate open to closed, but feels a bit ambiguous on its own. Thoughts?

![image](https://user-images.githubusercontent.com/29716057/85077741-d7ebc980-b177-11ea-9838-b3e90a57a48f.png)

**Comment by asangma** [comment]
Hmmm...I think with the itty bitty lashes it's ambiguous.

But that being said, and sorry to do this, but I think maybe we hold on this. 
I have been operating on the assumption that the two existing icons will be hard to distinguish in the interface. I should ensure we actually test this out with what we currently have.

I will get back to you.

**Comment by KatelynSeitz** [comment]
Sounds good. Thanks for the update, Alan.

**Comment by asangma** [comment]
| Interaction | What is the state? | What does the user need to know | What do we show? | Why? |
|------------|-------------------|-----------------------------------|--------------------|-------|
| Idle | Visible | It's visible. | No indicator | Visible is default.|
| Idle | Hidden | It's hidden. | Eyeball crossed<br/>Or "hidden" icon | Indicate hidden items. |
| Hover/focus | Visible | It's visible.<br/>It can be hidden. | Eyeball | Show current state. |
| Hover/focus | HIdden | It's hidden.<br/>It can be shown. | Eyeball crosssed<br/>Or "hidden" icon | Show current state. |
| Click/KB | Hidden | It's hidden.<br/>It can be shown. | Eyeball crossed<br/>Or "hidden" icon | Show current state. |
| Click/KB | Visible | It's visible.<br/>It can be hidden. | Eyeball | Show current state. |

After mocking up some interactions, it seems like during interaction, we should show **current** state. If we show the resulting state, the indicator will show one thing and then change when the item loses hover or focus. And this moment for the user can be confusing.

Showing the resulting state becomes more confusing because after clicking the toggle, the button stays focus, so after a user clicks the toggle, the indicator would continue to show the opposite of the current state of the item.

**Comment by johnmgriffith** [comment]
I think this description is spot on. Assuming that the object is visible, and not displaying the icon except when hovering over the card is a great choice.

The other additions I think are needed would be adding opacities to certain objects depending on their state or interaction. For example, hidden items are slightly greyed out, or on click the button gets just a little darker so the user has feedback that the state will change

**Comment by KatelynSeitz** [comment]
Makes sense! Thanks for the mapping these out. Would be great to see if the existing `view-visible` and `view-hidden` work well in the above interactions.
